tests-wtr/utils/highlight.test.ts

```diff
@@ -0,0 +1,139 @@
+            expect(styleTag).to.be.null; // Style tag should not be added if node.style is undefined
+        });
+
+        it('should handle elements without a pre-existing style attribute', () => {
+            highlightNode(testElement);
+            expect(testElement.hasAttribute('style')).to.be.true;
+            expect(testElement.__oldStyles).to.be.undefined; // No old styles to store
+        });
+    });
+
+    describe('unhighlightNode', () => {
+        it('should remove highlight styles and restore original styles if they existed', () => {
+            testElement.style.color = 'blue';
+            const originalCssText = testElement.style.cssText;
+            highlightNode(testElement);
+            expect(testElement.__oldStyles).to.equal(originalCssText);
+            const highlightedCssText = testElement.style.cssText;
+
+            unhighlightNode(testElement);
+
+            expect(testElement.style.cssText).to.equal(originalCssText);
+            expect(testElement.style.cssText).to.not.equal(highlightedCssText);
+            expect(testElement.__oldStyles).to.be.undefined;
+        });
+
+        it('should remove style attribute if no original styles were present', () => {
+            highlightNode(testElement);
+            expect(testElement.hasAttribute('style')).to.be.true;
+            expect(testElement.__oldStyles).to.be.undefined;
+
+            unhighlightNode(testElement);
+
+            expect(testElement.hasAttribute('style')).to.be.false;
+        });
+
+        it('should do nothing if node has no style property', () => {
+            const notAnElement = {} as any; // Using any here to test non-HTMLElement behavior
+            highlightNode(notAnElement);
+            unhighlightNode(notAnElement);
+            // No error should be thrown, and no properties should be changed or added.
+            expect(notAnElement.style).to.be.undefined;
+            expect(notAnElement.__oldStyles).to.be.undefined;
+        });
+
+        it('should do nothing if node has no style attribute (e.g., highlight was not called)', () => {
+            const initialAttributes = Array.from(testElement.attributes).map((attr) => attr.name);
+            unhighlightNode(testElement);
+            const finalAttributes = Array.from(testElement.attributes).map((attr) => attr.name);
+            expect(finalAttributes).to.deep.equal(initialAttributes);
+        });
+
+        it('should do nothing if node.style exists but no style attribute is set', () => {
+            Object.defineProperty(testElement, 'hasAttribute', { value: () => false, configurable: true });
+            const initialStyleCssText = testElement.style.cssText; // Should be empty
+
+            unhighlightNode(testElement);
+
+            expect(testElement.style.cssText).to.equal(initialStyleCssText);
+            expect(testElement.__oldStyles).to.be.undefined;
+        });
+    });
```

Comment:
What should happen if `highlightNode` is called multiple times on the same element or different elements without unhighlighting in between? Should we add a test for this?